### PR TITLE
Properly log smoothing errors

### PR DESCRIPTION
**Pull request by S34NW**
## What Does This PR Do
Makes these runtime, instead of throw a DD error, so people can actually know when they fail.

### CODE REVIEW

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

File: code/__HELPERS/icon_smoothing.dm [hunk 1]
```diff
@@ -153,7 +153,7 @@ DEFINE_BITFIELD(smoothing_junction, list(
 /atom/proc/smooth_icon()
 	smoothing_flags &= ~SMOOTH_QUEUED
 	if(!z) //nullspace are not sending their best
-		error("[type] called smooth_icon() without being on a z-level")
+		log_runtime(EXCEPTION("[src] called smooth_icon() without being on a z-level"))
```
[the remark this answers is not in the document]
Comment (S34NW) [reply code/__HELPERS/icon_smoothing.dm:156]: oh

[the remark this answers is not in the document]
Comment (S34NW) [reply code/__HELPERS/icon_smoothing.dm:156]: Okay, but don't merge this before yours

[the remark this answers is not in the document]
Comment (AffectedArc07) [reply code/__HELPERS/icon_smoothing.dm]: > Okay, but don't merge this before yours

But its safe to

[the remark this answers is not in the document]
Comment (S34NW) [reply code/__HELPERS/icon_smoothing.dm]: im smart

File: code/__HELPERS/icon_smoothing.dm [hunk 2]
```diff
@@ -153,7 +153,7 @@ DEFINE_BITFIELD(smoothing_junction, list(
 /atom/proc/smooth_icon()
 	smoothing_flags &= ~SMOOTH_QUEUED
 	if(!z) //nullspace are not sending their best
-		error("[type] called smooth_icon() without being on a z-level")
+		stack_trace("[src] called smooth_icon() without being on a z-level")
```
Comment (AffectedArc07) [inline code/__HELPERS/icon_smoothing.dm:156]:
```suggestion
		stack_trace("[type] called smooth_icon() without being on a z-level")
```

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**